pkg/scheduler/objects/queue.go

```diff
@@ -957,12 +957,10 @@ func (sq *Queue) addChildQueue(child *Queue) error {
 // This can be executed multiple times and is only effective the first time.
 // This is a noop on an unmanaged queue.
 func (sq *Queue) MarkQueueForRemoval() {
-	// need to lock for write as we don't want to add a queue while marking for removal
-	sq.Lock()
-	defer sq.Unlock()
+	children := sq.GetCopyOfChildren()
 	// Mark the managed queue for deletion: it is removed from the config let it drain.
 	// Also mark all the managed children for deletion.
-	if sq.isManaged {
+	if sq.IsManaged() {
 		log.Log(log.SchedQueue).Info("marking managed queue for deletion",
 			zap.String("queue", sq.QueuePath))
 		if err := sq.handleQueueEvent(Remove); err != nil {
```

Comment:
we don't hold lock now, but `handleQueueEvent` can modify inner variable (https://github.com/apache/yunikorn-core/blob/master/pkg/scheduler/objects/queue.go#L557)

Reply:
Good catch... gotta figure this out.

Reply:
Done